app/realty/domain/floor/selector.py

```diff
@@ -0,0 +1,46 @@
+from django.core.exceptions import ObjectDoesNotExist, MultipleObjectsReturned
+from django.db.models import Count
+
+from realty.domain.flat.entities import FlatEntity
+from realty.domain.floor.entities import FloorEntity
+from realty.models.floor import Floor
+
+
+class FloorSelector:
+    @staticmethod
+    def get_floors_with_total_flats():
+        floors = Floor.objects.annotate(total_flats=Count('flat'))
+        return floors
+
+    @staticmethod
+    def get_floor_detail(pk):
+        try:
+            floor = Floor.objects.prefetch_related('flat_set').get(id=pk)
```

Comment:
Здесь лучше как-нибудь так все сущности подтянуть
```suggestion
            floor = Floor.objects.prefetch_related('flat_set__floor', 'flat_set__category', ...).get(id=pk)
```